### ISSUE TITLE
explicit transclusion at point without #+transclude and in middle of a line.

### ISSUE THREAD

**Issue by natask** [nobiot/org-transclusion#43]
My use case is as follows, I would like to transclude links that I have built up in the past. I have links that I have placed either as headline text or plain lists. I would like to still transclude these  links explicitly through transclusion at point. The current implementation requires #+transclude and doesn't work for links that aren't the first text in a line. 

``` org
* [[link]]
:properties:
:end:

- [[link]]
```
I would like to explicitly transclude them as follows

``` org
* [[link]]
:properties:
:end:
stuff from [[link]]

- [[link]]
stuff from [[link]]
```

the link can also be in the middle of text and transclusion would still work. There is no ambiguity because it is explicit transclusion. If need to explicitly transclude the whole buffer, can either innumerate all links and transclude them adding appropriate space or translucde only the first link on a line. My use case doesn't fall here so I don't have great many things to comment. 
``` org
* this is a [[link]] is it not
:properties:
:end:
stuff from [[link]]

- this is a [[link]] is it not
stuff from [[link]]
```

**Comment by natask** [comment]
I see myself implementing this in the next couple of weeks because It is useful to me.

**Comment by nobiot** [comment]
Cool! 

For Org links, if you advise or override `org-transclusion--ok-to-transclude`, and return `t` or `nil` for your conditions, the rest of the facilities should take care of the transclusion -- if you would like to use Org-transclusion as the starting point, that is.

I have not tested this, so there might be some tweaks required, but that is the intention I had.

We could even consider using customizing variable (list of function names, for example) to allow for custom transclusion determination logic (ok or not to transclude).

**Comment by nobiot** [comment]
I just realized that I put `(bolp)` as part of the `and` condition. 
I will need to move this condition inside `org-transclusion--ok-to-transclude`, so that the idea of advising/overriding it will work. 

https://github.com/nobiot/org-transclusion/blob/main/org-transclusion.el#L771-L773

Let me know if you are considering to use Org-transclusion as the basis for your implementation; if so, I will do the change.

**Comment by natask** [comment]
yes, I will base it on org-transclusion.

**Comment by natask** [comment]
I have it on https://github.com/natask/org-transclusion/tree/feat/transclude-explicitly. works even if the link isn't the beginning of a sentence. puts transclusion before the next org element. I don't use #+transclude: t (I manually transclude as needed) and it would be more work to support both (manage state when removing overlays for #+transclude: t and manual) so I have removed it temporarily from my repo.

**Comment by nobiot** [comment]
@natask 

FYI.
Wondered if you were interested in looking at what I started cooking.

Working on a major, breaking change (summarised in PR #69) on a new branch v0.1.0.

Among other things, I am testing an idea to use only `#+transclude:` keyword in this form: `#+transclude: t "path/to/link.ext" :level n`.

With this change, I have also attempted to incorporate your idea of a link in a headline.

The idea is... if you have this:
```
* [[link]]
:properties:
:end:
```
You can create a transclusion keyword from a link at point with [`org-transclusion-create-from-link`](https://github.com/nobiot/org-transclusion/blob/dev/0.1.0/org-transclusion.el#L377). 

At the moment, I struggle to define where the transclusion keyword should be placed (`org-forward-element`, which I believe yo use, does not seem to be the best).

State management is still not great but I am working on it too.

**Comment by natask** [comment]
instead of `org-forward-element`, which as you say doesn't work on the last headline, I was looking to write dwim version that places the trascluded element at the next line in the case of a line or plainlist or below the last drawer in the case of a headline.

**Comment by nobiot** [comment]
I see. That sounds cool. 
I was simply thinking to craw down each line to find `bolp = eolp`, and add a blank line for above and bottom to make room 😉 Too crude, perhaps but might be enough to start with.

**Comment by nobiot** [comment]
Just added an initial implementation of `create-from-link` function bd3d1b0.
I would like to see where it will lead us.

**Comment by nobiot** [comment]
Merged v0.1.0 with the main branch. I have decided to streamline the syntax based on your approach. Thank you.
Now transclusion is done on a `#+transclude:` keyword only, which you can make out of a link. 

I have enabled live-sync edit in this version as an experiment. You can find more about the new appoach in the new README.

I would appreciate it if you could try the new version and provide any feedback. Closing for now. Thank you.
